co-simulation/fed/mosaic-carma/src/main/java/org/eclipse/mosaic/fed/carma/ambassador/CarmaRegistrationReceiver.java

```diff
@@ -64,7 +64,7 @@ public void run() {
             }
 
             // parse message
-            String receivedPacket = new String(msg.getData());
+            String receivedPacket = new String(msg.getData(), 0, msg.getLength());
```

Comment:
Was halfway through writing up a question as to why you chose to use this version of the constructor before I realized the error you were correcting, very nice catch! Previously it was encoding the whole buf[] contents even if the received packet was smaller than `UDP_MTU` bytes right?

Reply:
I found this issue when I was running the `InfrastructurethereRegistrationReceiverTest.java` which used same way to getting the packet and convert that to the specified class format. 
When we created the `String` object from the received bytes, it may contain extra characters at the end that were not part of the original message. This is because the `buf` array used in the `DatagramPacket` constructor is preallocated and may contain unused bytes. Therefore, the String object created from `msg.getData()` may have trailing null characters. 
This null characters causes an exception when `gson.fromJson` trying to convert received packet into the specified class format. So I modified the `String` object using the `String` constructor that takes a byte array and the length of the message to ensure that only the actual message bytes are used to create the `String`